store/src/java/com/zimbra/cs/service/FileUploadServlet.java

```diff
@@ -161,11 +161,12 @@ public static final class Upload {
             } else {
                 mimeType = getMimeType(file);
                 contentType = mimeType.toString();
-
+                extension = filename == null ? "" : FilenameUtils.getExtension(filename).trim();
```

Comment:
have we checked why tika is not able to detect p7m content type correctly? why are we putting hacks in our code when third party library should be able to handle this scenario
check https://github.com/apache/tika/blob/main/tika-core/src/main/resources/org/apache/tika/mime/tika-mimetypes.xml#L790

Reply:
If for some reason its failing to detect the mime type from library, we can try and add mime type into https://github.com/Zimbra/zm-mailbox/blob/develop/store-conf/conf/custom-mimetypes.xml

Reply:
This seems to be an open issue with Tika. Please find below resource for more details - 
https://issues.apache.org/jira/browse/TIKA-1997